packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -340,10 +339,20 @@ class FlutterPlugin implements Plugin<Project> {
                         "packages", "flutter_tools", "gradle", "src", "main", "kotlin",
                         "dependency_version_checker.gradle.kts")
                 project.apply from: dependencyCheckerPluginPath
-            } catch (Exception ignored) {
-                project.logger.error("Warning: Flutter was unable to detect project Gradle, Java, " +
-                        "AGP, and KGP versions. Skipping dependency version checking. Error was: "
-                        + ignored)
+            } catch (Exception e) {
+
+                if (!project.failedDependencyChecks) {
```

Comment:
This if statement is a bit difficult to read since you are negating "failed" then logging an error. 
```suggestion
                // Exception means something went wrong. 
                if (project.failedToCheckDependencies) {
                   // Possible bug in dependency checking code warn and do not block build.  
                   ...
                } else {
                  // If failedToCheckDependencies is unset or false then the exception is thrown by us and/or an an error that should block the build. 
                  throw e. 
                }
```

Reply:
Made the property a constant, and changed it to `usesUnsupportedDependencyVersions` so that the check reads better as:
```
if (!project.usesUnsupportedDependencyVersions) {
...
} else {
...
}
```